Library/Homebrew/development_tools.rb

```diff
@@ -76,7 +76,7 @@ def llvm_clang_build_version
 
     def non_apple_gcc_version(cc)
       (@non_apple_gcc_version ||= {}).fetch(cc) do
-        path = HOMEBREW_PREFIX/"opt/gcc/bin"/cc
+        path = HOMEBREW_PREFIX/"opt/#{CompilerSelector.preferred_gcc}/bin"/cc
```

Comment:
This line was originally added in Homebrew 0.9.8, seven years ago!
https://github.com/Homebrew/brew/commit/55d277c33592f922f506aa2e07c0f81c6d33fd74#diff-b5752027e688b330ed260dd944dbf2674c98c1b44355282d93ead10f89dbc3fdR171